usr/share/rear/lib/layout-functions.sh

```diff
@@ -751,52 +751,51 @@ function get_part_device_name_format() {
 # (usually disk_mappings file in $VAR_DIR).
 function apply_layout_mappings() {
 
-    # apply_layout_mappings need one argument (a non-empty file which contains disk device to migrate).
-    if [ -z $1 ] ; then
-        BugError "apply_layout_mappings function called without argument (file_to_migrate)."
-    else
-        file_to_migrate="$1"
-    fi
+    # apply_layout_mappings need one argument.
+    [ "$1" ] || BugError "apply_layout_mappings function called without argument (file_to_migrate)."
 
     if [ -z "$MIGRATION_MODE" ] ; then
         return 0
     fi
 
-    # We temporarily map all devices in the mapping to new names _REAR[0-9]+_
-    replaced_count=0
-    replacement_file="$TMP_DIR/replacement_file"
-    : > "$replacement_file"
-
-    # Generate replacements.
-    while read source target junk ; do
-        if ! has_replacement "$source" ; then
-            add_replacement "$source"
-        fi
+    # Only apply layout mapping on non-empty file.
+    if [ -s "$1" ] ; then
+        # We temporarily map all devices in the mapping to new names _REAR[0-9]+_
+        replaced_count=0
+        replacement_file="$TMP_DIR/replacement_file"
+        : > "$replacement_file"
+
+        # Generate replacements.
```

Comment:
Suggestion for a comment that summarizes what took me some time to realize:
```
# Generate unique words as replacement placeholders to correctly handle circular replacements (e.g. sda -> sdb and sdb -> sda).
# Replacement strategy is
# 1) replace all source devices with a unique word (the "replacement")
# 2) replace all unique replacement words with the target device
```

Reply:
Yep, you are right. It also took me a lot of time to understand this.